lib/auth/auth.ts

```diff
@@ -145,16 +145,12 @@ function doAuth(
     }
 
     // Corner cases managed, we're left with normal auth
-    // TODO What's happening here?
-    // @ts-ignore
-    res.params.log = log;
-    if (res.params.version === 2) {
-        // @ts-ignore
-        return vault!.authenticateV2Request(res.params, requestContexts, cb);
+    res.params!.log = log;
```

Comment:
do you know why Typescript is not able to infer the type here?
As far as I understand, `extractParams()` return either AuthV2RequestParams, AuthV4RequestParams or AuthInfo. Since the AuthInfo case is handle, TS should be able to deduce this will be either AuthV2RequestParams or AuthV4RequestParams...

Maybe this is due to the "optional" params (`params?`) in result of extractParams?
And the return type should be better type to either return an error without params, or params with null error: so that type deduction can work its magic, and we don't need type coercion...

Reply:
I updated the PR, I created a new type that allows TS to understand that params is defined if `err` is `null`.